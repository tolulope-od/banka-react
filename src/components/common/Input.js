import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ type, name, placeholder, id, value, onChange, required, disabled }) => {
  return (
    <input
      type={type}
      name={name}
      placeholder={placeholder}
      id={id}
      value={value}
      onChange={onChange}
      required={required}
      disabled={disabled}
    />
  );
};

Input.propTypes = {
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  required: PropTypes.bool,
  disconnect: PropTypes.bool
};

export default Input;
