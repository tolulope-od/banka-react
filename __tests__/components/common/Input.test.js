import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../../../src/components/common/Input';

describe('<Input />', () => {
  it('Renders the Input component', () => {
    const wrapper = shallow(
      <Input
        type="email"
        name="email"
        placeholder="Email"
        id="user-email"
        value="sample@email.com"
        onChange={jest.fn()}
        required={true}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
