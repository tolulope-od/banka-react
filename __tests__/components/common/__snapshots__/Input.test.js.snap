// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Renders the Input component 1`] = `
<input
  id="user-email"
  name="email"
  onChange={[MockFunction]}
  placeholder="Email"
  required={true}
  type="email"
  value="sample@email.com"
/>
`;
